File: 190-reverse-bits/reverse-bits.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t right= 1, left= 1L<<31, ans=0;
        // left= left<<31;
        while(left>right){
            if((left&n)>0){
                ans|=right;
            }
            if((right&n)>0){
                ans|=left;
            }
            left/=2;
            right*=2;
        }
        if((left&n)>0){
            ans|=right;
        }
        return ans;
    }
};

File: 190-reverse-bits/reverse-bits.java
public class Solution {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        int r=0, l=31, ans=0;
        while(l>r){
            int right= 1<<r, left= 1<<l;
            if((left&n)!=0){
                ans+= right;
            }
            if((right&n)!=0){
                ans+=left;
            }
            l--;
            r++;
        }
        if(l==r) ans+=(1<<l);
        return ans;

    }
}